source/agora/node/Ledger.d

```diff
@@ -358,10 +358,10 @@ public class Ledger
         auto reason = tx.isInvalidReason(this.utxo_set.getUTXOFinder(),
             expected_height, &this.payload_checker.check);
 
-        if (reason !is null || !this.pool.add(tx))
+        if (tx.type == TxType.Coinbase || reason !is null || !this.pool.add(tx))
```

Comment:
It would be better to skip it before tx validation. Otherwise this might become a DDos vector.